RecoLocalTracker/SiPixelClusterizer/plugins/gpuClusterChargeCut.h

```diff
@@ -119,8 +126,21 @@ namespace gpuClustering {
 
       // renumber
       __shared__ uint16_t ws[32];
-      cms::cuda::blockPrefixScan(newclusId, nclus, ws);
-
+      constexpr auto maxThreads = 1024;
+      auto minClust = nclus > maxThreads ? maxThreads : nclus;
+
+      cms::cuda::blockPrefixScan(newclusId, newclusId, minClust, ws);
+      if constexpr (maxNumClustersPerModules > maxThreads)  //only if needed
+      {
+        //TODO: most probably there's a smarter implementation for this
+        if (nclus > maxThreads) {
+          cms::cuda::blockPrefixScan(newclusId + maxThreads, newclusId + maxThreads, nclus - maxThreads, ws);
```

Comment:
doesn't think need a check that `nclus <= 2 * maxThreads` ?

Reply:
alternatively, the `if` could be changed into a `for` loop, with as many iterations as needed to cover `nclus` in blocks of `maxThreads`

Reply:
Right! Going for a loop (seems more robust).